website/src/components/canvas/Sidebar.jsx

```diff
@@ -37,16 +52,24 @@ class Sidebar extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
-      opacity: .7
+      opacity: .7,
     }
     this.mouseEnter = this.mouseEnter.bind(this);
     this.mouseLeave = this.mouseLeave.bind(this);
+    this.onClickOpen = this.onClickOpen.bind(this);
+    this.onClickClose = this.onClickClose.bind(this);
     this.removeElement = this.removeElement.bind(this);
     this.addElement = this.addElement.bind(this);
     this.mapOptionToDiv = this.mapOptionToDiv.bind(this);
     this.listItems = CC.SIDEBAR_BUTTONS.map(this.mapOptionToDiv);
-  }
 
+    }
+    componentDidMount(){
```

Comment:
This needs a docstring. 
You should also probably be initializing the isOpen and closed states in the constructor